src/choreo/tower/fd_tower.c

```diff
@@ -231,20 +231,22 @@ fd_tower_threshold_check( fd_tower_t const *    tower,
 
   /* Iterate all the vote accounts. */
 
-  fd_voter_t const * epoch_voters = fd_epoch_voters_const( epoch );
-  for (ulong i = 0; i < fd_epoch_voters_slot_cnt( epoch_voters ); i++ ) {
-    if( FD_LIKELY( fd_epoch_voters_key_inval( epoch_voters[i].key ) ) ) continue /* most slots are empty */;
+  for (ulong i = 0; i < vote_states_len; i++ ) {
+    fd_replay_out_vote_state_t const * vote_state = &vote_states[i];
 
-    fd_voter_t const * voter = &epoch_voters[i];
+    fd_voter_t * epoch_voters = fd_epoch_voters( epoch );
+    fd_voter_t * voter        = fd_epoch_voters_query( epoch_voters, vote_state->key, NULL );
+    if( FD_UNLIKELY( !voter ) ) {
+      /* TODO: handle this case by adding a vote account to the epoch voters */
+      FD_LOG_DEBUG(( "[%s] voter %s was not in epoch voters", __func__,
```

Comment:
what does the TODO mean? also should at least warn here, if not crit? i would expect replay to always pass in data that is sync with tower in the new copy-based model